Important fixes to build_materialx_osl.py

amartinez-cg opened the pull request:
previously script was overwriting .osl files

added some checking to make sure source and destination exist

corrected osl file counter

commented functions


## Description

<!-- Please provide a description of what this PR is meant to fix, and  -->
<!-- how it works (if it's not going to be very clear from the code).   -->

## Tests

<!-- Did you / should you add a testsuite case (new test, or add to an  -->
<!-- existing test) to verify that this works?                          -->


## Checklist:

<!-- Put an 'x' in the boxes as you complete the checklist items -->

- [ ] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [ ] My code follows the prevailing code style of this project.

lgritz commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -15,14 +15,16 @@
 
 OSLC_CMD = 'oslc '
-OSLC_PATH = ''
+OSLC_PATH = '../../dist/linux64/bin/oslc'
```
You can't have this hard-coded to the architecture

amartinez-cg commented:
Not sure what else to do about the architecture path that provide a global variable that can be changed by hand.

lgritz commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -14,15 +14,19 @@
+ARCH = 'linux64'
 MX_SOURCE = '../shaders/MaterialX'
-OSL_DEST = '../../dist/linux64/shaders/MaterialX'
+OSL_DEST = '../../dist/%s/shaders/MaterialX'%ARCH
```
This should probably be 'build', not 'dest'.
The 'make install' should copy things from build to dest, but none of the build process should put things directly in dest.

amartinez-cg commented:
10-4!

On Thu, Jun 8, 2017 at 10:51 AM, Larry Gritz <notifications@github.com>
wrote:

> *@lgritz* commented on this pull request.
> ------------------------------
>
> In src/build-scripts/build_materialX_osl.py
> <https://github.com/amartinez-cg/OpenShadingLanguage/pull/13#discussion_r120955064>
> :
>
> >  MX_SOURCE = '../shaders/MaterialX'
> -OSL_DEST = '../../dist/linux64/shaders/MaterialX'
> +OSL_DEST = '../../dist/%s/shaders/MaterialX'%ARCH
>
> This should probably be 'build', not 'dest'.
> The 'make install' should copy things from build to dest, but none of the
> build process should put things directly in dest.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/amartinez-cg/OpenShadingLanguage/pull/13#pullrequestreview-42964865>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABFtX6enKDK_TG_XWItWVbHORZ1Iv3BXks5sCDSxgaJpZM4N0V89>
> .
>

lgritz commented on src/build-scripts/build_materialX_osl.py:
```diff
@@ -14,15 +14,19 @@
+import argparse
 
+# global defaults
+ARCH = 'linux64'
```
I'm thinking that this python script should take the architecture (or, even better, just the paths to oslc and the shaders/MaterialX directory), which CMakeLists.txt can easily pass to it.

If the oslc path is not supplied, it should just assume 'oslc' is in the execution path, and if the build/shaders area isn't supplied, it should just assume the current dir.

lgritz replied:
I mean "take" -> as parameters to this script

amartinez-cg commented:
This latest commit does that, but keeps the global declarations, which is
confusing.  I think Ill get rid of the globals and pass the arguments
through the functions.

On Thu, Jun 8, 2017 at 10:54 AM, Larry Gritz <notifications@github.com>
wrote:

> *@lgritz* commented on this pull request.
> ------------------------------
>
> In src/build-scripts/build_materialX_osl.py
> <https://github.com/amartinez-cg/OpenShadingLanguage/pull/13#discussion_r120955648>
> :
>
> >
> +# global defaults
> +ARCH = 'linux64'
>
> I mean "take" -> as parameters to this script
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/amartinez-cg/OpenShadingLanguage/pull/13#discussion_r120955648>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABFtX0GvDWgd4hw_-DKla6cTVBMMkaxlks5sCDVFgaJpZM4N0V89>
> .
>

amartinez-cg committed:
Added type_string for filename suffixes,
previously script was overwriting .osl files

added some checking to make sure source and destination exist

corrected osl file counter

commented functions

Added arguments for architecture and verbosity.

Added more arguments for oslc compiling.  Had to add back header installation in CMakeLists.txt
